airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/pipeline.py

```diff
@@ -273,6 +291,22 @@ def create_connector_report(results: List[StepResult]) -> ConnectorReport:
                 metadata_upload_results = await metadata_upload_step.run()
                 results.append(metadata_upload_results)
 
+            # Try to convert the context to a PyPIPublishContext. If it returns None, it means we don't need to publish to pypi.
```

Comment:
Currently the behavior is like this:
* If pypi is not enabled, the steps are not even shown
* If pypi is enabled but it's published already, the steps are shown as skipped

My reasoning was that it might be weird / unexpected to even show pypi-related stuff on connector reports that have nothing to do with it. Basically assuming the following semantic meaning of skip: "Normally this step would be performed, but an edge case prevent us from doing so - you might look into that".

No strong opinion on this though, we can also always show them as "skipped" if you think it makes sense.

Reply:
Agreed, it makes no sense to show skipped pypi steps for a java connector.